src/org/ods/openshift/OpenShiftResourceMetadata.groovy

```diff
@@ -0,0 +1,462 @@
+                )
+            }
+            project = "${context.projectId}-${config.environment}"
+        }
+        return project
+    }
+
+    /**
+     * Tries to guess the role of the component from the quickstarter name.
+     * This can be done only when provisioning the component for the first time, not when redeploying.
+     * However, when redeploying, the corresponding labels are expected to be already set,
+     * and the role will only change if explicitly set up by the user.
+     *
+     * It can only guess the role for quickstarters following the naming convention of using prefixes related
+     * to the nature of the component: 'be-' for backend, 'fe-' for frontend, 'de-' for data science.
+     *
+     * @return the role of the component or <code>null</code>, if it could not be guessed.
+     */
+    private guessRoleFromQuickStarterName() {
+        def quickStarterName = context.sourceDir
+        if (!quickStarterName) {
+            return null
+        }
+        if (quickStarterName.startsWith('fe-')) {
+            return ROLE_FRONTEND
+        }
+        if (quickStarterName.startsWith('ds-')) {
+            return ROLE_SUBSYSTEM
+        }
+        if (quickStarterName.startsWith('be-') && !quickStarterName.startsWith('be-fe-')) {
+            return ROLE_BACKEND
+        }
+        return null
+    }
+
+    /**
+     * Reads the metadata from an existing component.
+     *
+     * @return a <code>Map</code> with the metadata set up in the component.
+     */
+    private getMetadataFromComponent() {
+        return readMetadata('metadata.yml')
+    }
+
+    /**
+     * Reads the default metadata for components provisioned from the current quickstarter
+     * from the quickstarter repository.
+     *
+     * @return a <code>Map</code> with the default metadata for components provisioned with the current quickstarter.
+     */
+    private getMetadataFromQuickStarter() {
+        return readMetadata("${context.componentId}/metadata.yml")
+    }
+
+    /**
+     * Searches for a helm main chart file in the component.
+     * If found, it determines the chart name and version string,
+     * with format name-version.
+     * It replaces any '+' symbol in the version with a '_' symbol.
+     *
+     * @return the chart name-version string, or null if no chart file can be found.
+     */
+    private getChartNameAndVersion() {
+        if (config.chartDir) {
+            def chartPath = "${config.chartDir}/Chart.yaml"
+            if (steps.fileExists(chartPath)) {
+                def chart = steps.readYaml(file: chartPath)
+                // The following replacement is as per the specification of this label
+                // and independent of the way we choose to sanitize labels otherwise.
+                return "${chart.name}-${chart.version.replace('+' as char, '_' as char)}"
+            }
+        }
+        return null
+    }
```

Comment:
Why do we read from chart here? why not let that be handled by Helm itself?

Reply:
I considered the possibility to integrate with the helm tool and even get several label values directly it, but I left that as out of scope (actually, setting this helm-related chart is an extra).
This was so easy to do, that it felt more straightforward just to read the chart file as a YAML. It can be improved in the future in quite a tranparent way, I would say.
If you don't feel confortable with my approach, I can disable this label for the moment or see how long it would take to integrate with the helm tool.